clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6000,6 +6000,12 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back(A->getValue());
   }
 
+  if (Args.hasArg(options::OPT_fsycl) &&
+      Args.hasArg(options::OPT_ffreestanding)) {
+    D.Diag(diag::err_drv_option_conflict) << "-fsycl"
+                                          << "-ffreestanding";
+  }
```

Comment:
I believe this code should be moved into `Driver::CreateOffloadingDeviceToolChains()` - the conflicting SYCL options are checked around line 867 of `Driver.cpp`.

Reply:
@AGindinson thank you for the review, yes moved this code to Driver.cpp